lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -2888,6 +2888,108 @@ bool LowererMDArch::GenerateFastAnd(IR::Instr * instrAnd)
     return true;
 }
 
+bool LowererMDArch::GenerateFastDiv(IR::Instr * instrDiv)
+{
+    Assert(instrDiv);
+    IR::Opnd * divisor = instrDiv->GetSrc2(); // denominator
+
+    if (PHASE_OFF(Js::BitopsFastPathPhase, this->m_func) || !divisor->IsIntConstOpnd() ||
+        !(instrDiv->m_opcode == Js::OpCode::Div_I4 || instrDiv->m_opcode == Js::OpCode::Rem_I4)) //ToDo Optimize unsigned division
+    {
+        return false;
+    }
+
+    IR::Opnd* divident = instrDiv->GetSrc1(); // nominator
+    IR::Opnd* dst      = instrDiv->GetDst();
+    int constDivisor   = divisor->AsIntConstOpnd()->AsInt32();
+    bool isNegDevisor  = false;
+
+    if (constDivisor < 0)
+    {
+        isNegDevisor = true;
+        constDivisor = abs(constDivisor);
+    }
+
+    if (constDivisor < 2 || constDivisor > INT32_MAX - 1)
+    {
+        return false;
+    }
+
+    if (Math::IsPow2(constDivisor)) // Power of two
+    {
+        // Negative dividents needs the result incremented by 1
+        // Following sequence avoids branch
+        // For q = n/d and d = 2^k
+
+        //      sar q n k-1   //2^(k-1) if n < 0 else 0
+        //      shr q q 32-k
+        //      add q q n
+        //      sar q q k
+
+        int k = Math::Log2(constDivisor);
+        Lowerer::InsertShift(Js::OpCode::Shr_A, false, dst, divident, IR::IntConstOpnd::New(k - 1, TyInt8, this->m_func, true), instrDiv);
```

Comment:
>IR::IntConstOpnd::New(k - 1, TyInt8, this->m_func, true [](start = 70, length = 55)

You still have constants which are user controlled where you pass dontEncode as true.  Even though they are small constant, you should let the security code decide to encode them or not.

Reply:
thanks. fixed.